pymtl3/stdlib/ifcs/CalleeIfc.py

```diff
@@ -0,0 +1,167 @@
+#=========================================================================
+# CalleeIfc.py
+#=========================================================================
+# RTL method interface. RTL equivalence of NonBlockingCalleeIfc in CL
+#
+# Author : Yixiao Zhang
+#   Date : June 10, 2019
+
+from __future__ import absolute_import, division, print_function
+
+from copy import deepcopy
+
+from pymtl3 import *
+from pymtl3.dsl.errors import InvalidConnectionError
+
+
+#-------------------------------------------------------------------------
+# CalleeRTL2CL
+#-------------------------------------------------------------------------
+class CalleeRTL2CL( Component ):
+
+  def construct( s, ifc_rtl_callee ):
+
+    ifc_rtl_caller = deepcopy( ifc_rtl_callee )
+    ifc_rtl_caller._dsl.constructed = False
+    s.ifc_rtl_caller = ifc_rtl_caller.inverse()
+
+    s.called = False
+    s.rdy = False
+
+    # generate upblk depending on args
+    if s.ifc_rtl_caller.args:
```

Comment:
Same here. If you pass in ArgsType and RetType you can check if ArgsType is None.